core/templates/dev/head/pages/exploration_player/AudioControlsDirective.js

```diff
@@ -64,12 +65,19 @@ oppia.directive('audioControls', [
           };
 
           $scope.playPauseAudioTranslation = function() {
-            // TODO(tjiang11): Change from on-demand loading to pre-loading.
-
-            // TODO(tjiang11): On first play, ask learner to pick language
-            // and subsequently for confirmation to use bandwidth 
-            // to download audio files.
+            if (!AudioPreloaderService.hasPreloaded()) {
+              if (getCurrentAudioTranslation()) {
```

Comment:
Because I suppose it'd be better to be extra cautious if they're being restrictive on their bandwidth--?